pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -185,11 +200,27 @@ def on_pretrain_routine_start(self, trainer, pl_module):
         self.__resolve_ckpt_dir(trainer)
         self.save_function = trainer.save_checkpoint
 
+    def on_train_batch_end(self, trainer, pl_module, *args, **kwargs) -> None:
+        """ Save checkpoint on train batch end if we meet the criteria for `every_n_train_steps` """
+        if self._should_skip_saving_checkpoint(trainer):
+            return
+        step = trainer.global_step
+        skip_batch = self._every_n_train_steps < 1 or ((step + 1) % self._every_n_train_steps != 0)
+        if skip_batch:
+            return
+        self.save_checkpoint(trainer, pl_module)
```

Comment:
```suggestion
        self.save_checkpoint(trainer)
```
otherwise it prints this warning:
`
DeprecationWarning: ModelCheckpoint.save_checkpoint signature has changed in v1.3. The pl_module parameter has been removed`

cc @carmocca